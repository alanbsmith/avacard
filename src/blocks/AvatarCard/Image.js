import styled from 'styled-components';
import PropTypes from 'prop-types';

import { dimensions } from '../../utils';

const Image = styled.img.attrs({
  src: props => `https://api.adorable.io/avatars/150/${props.email}.png`,
  alt: 'adorable user avatar',
})`
  border-radius: ${dimensions('borderRadius')};
  height: 106px;
  width: 106px;
`;

Image.propTypes = {
  email: PropTypes.string.isRequired,
};

export default Image;
