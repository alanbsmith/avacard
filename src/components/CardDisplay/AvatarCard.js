import React from 'react';
import PropTypes from 'prop-types';

import AvatarCardBlock from '../../blocks/AvatarCard';
import Grid from '../../blocks/Grid';

const { Column, Container, Row } = Grid;

function AvatarCard({ catchPhrase, email, name }) {
  return (
    <AvatarCardBlock tabIndex="0">
      <Row>
        <Column modifiers={['alignCenter']}>
          <AvatarCardBlock.Image email={email} />
        </Column>
        <Column>
          <Container modifiers={['fluid']}>
            <Row>
              <AvatarCardBlock.Text modifiers={['italicize']}>
                "{catchPhrase}"
              </AvatarCardBlock.Text>
            </Row>
            <Row>
              <Column modifiers={['flexColumn', 'fluid']}>
                <AvatarCardBlock.Text modifiers={['semiBold']}>
                  {name}
                </AvatarCardBlock.Text>
                <AvatarCardBlock.Text modifiers={['lowercase']}>
                  {email}
                </AvatarCardBlock.Text>
              </Column>
            </Row>
          </Container>
        </Column>
      </Row>
    </AvatarCardBlock>
  );
}

AvatarCard.propTypes = {
  catchPhrase: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default AvatarCard;
