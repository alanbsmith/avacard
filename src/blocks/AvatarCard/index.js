import { colors, gridScale } from "../../utils";

import Text from '../../elements/Text';
import Card from '../Card';
import Image from './Image';

const AvatarCard = Card.extend`
  flex: 0 0 ${gridScale(35)};
  height: 206px;
  padding: ${gridScale(4)} ${gridScale(1)};
  transition: border-color 300ms ease;

  &:hover,
  &:focus {
    border-color: ${colors('chrome200')};
  }
`;

AvatarCard.Image = Image;
AvatarCard.Text = Text;

export default AvatarCard;
