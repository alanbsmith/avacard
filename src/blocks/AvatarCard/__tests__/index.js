import { generateSnapshots } from "__tests__/helpers";

import AvatarCard from '../index';

generateSnapshots(AvatarCard);
