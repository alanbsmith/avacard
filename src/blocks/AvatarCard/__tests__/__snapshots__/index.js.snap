// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled.div renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  border-radius: 3px;
  border: solid 1px #EEF1F2;
  box-shadow: 3px 3px rgba(238,241,242,0.47);
  box-sizing: border-box;
  margin: 0.625rem;
  -webkit-flex: 0 0 21.875rem;
  -ms-flex: 0 0 21.875rem;
  flex: 0 0 21.875rem;
  height: 206px;
  padding: 2.5rem 0.625rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
}

.c0:hover,
.c0:focus {
  border-color: #BAC3CB;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply alignCenter modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
  border: solid 1px #EEF1F2;
  box-shadow: 3px 3px rgba(238,241,242,0.47);
  box-sizing: border-box;
  margin: 0.625rem;
  -webkit-flex: 0 0 21.875rem;
  -ms-flex: 0 0 21.875rem;
  flex: 0 0 21.875rem;
  height: 206px;
  padding: 2.5rem 0.625rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
}

.c0:hover,
.c0:focus {
  border-color: #BAC3CB;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply flexColumn modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 3px;
  border: solid 1px #EEF1F2;
  box-shadow: 3px 3px rgba(238,241,242,0.47);
  box-sizing: border-box;
  margin: 0.625rem;
  -webkit-flex: 0 0 21.875rem;
  -ms-flex: 0 0 21.875rem;
  flex: 0 0 21.875rem;
  height: 206px;
  padding: 2.5rem 0.625rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
}

.c0:hover,
.c0:focus {
  border-color: #BAC3CB;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply fluid modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  padding: 0;
  border-radius: 3px;
  border: solid 1px #EEF1F2;
  box-shadow: 3px 3px rgba(238,241,242,0.47);
  box-sizing: border-box;
  margin: 0.625rem;
  -webkit-flex: 0 0 21.875rem;
  -ms-flex: 0 0 21.875rem;
  flex: 0 0 21.875rem;
  height: 206px;
  padding: 2.5rem 0.625rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
}

.c0:hover,
.c0:focus {
  border-color: #BAC3CB;
}

<div
  className="c0"
/>
`;
